packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -144,12 +144,12 @@ class AnalyzeCommand extends FlutterCommand {
 
   @override
   Future<FlutterCommandResult> runCommand() async {
-    final bool? suggestionFlag = boolArg('suggestions');
-    final bool machineFlag = boolArg('machine') ?? false;
-    if (suggestionFlag != null && suggestionFlag == true) {
+    final bool suggestionFlag = boolArg('suggestions');
+    final bool machineFlag = boolArg('machine');
+    if (suggestionFlag) {
       final String directoryPath;
-      final bool? watchFlag = boolArg('watch');
-      if (watchFlag != null && watchFlag) {
+      final bool watchFlag = boolArg('watch');
+      if (watchFlag) {
```

Comment:
Nit: consider removing `watchFlag` and moving boolArg('watch') into the `if`

Reply:
Could also do the same with `suggestionFlag` and `machineFlag` since these are all referenced only once

Reply:
Done, thanks!